### PR TITLE
added type hints and docstrings

### PR DESCRIPTION
**Pull request by joao-fm-santos**
Adding type hints and will be adding descriptions here!

https://github.com/MrPowers/mack/issues/47

### CODE REVIEW

**Comment by MrPowers** [comment]
@joao-fm-santos - thanks for the PR.  Can you please take a look @robertkossendey / @souvik-databricks?

File: mack/__init__.py [hunk 1]
```diff
@@ -311,7 +425,23 @@ def append_without_duplicates(
     ).whenNotMatchedInsertAll().execute()
 
 
-def is_composite_key_candidate(delta_table: DeltaTable, cols: List[str]) -> bool:
+def is_composite_key_candidate(delta_table: DeltaTable, cols: List) -> bool:
```
Comment (robertkossendey) [inline mack/__init__.py:428]: @joaopfmsantos why did you remove the `[str]` hint here?

**Comment by robertkossendey** [comment]
@joao-fm-santos thank you for the PR! Looks great :) just one minor comment

**Comment by MrPowers** [comment]
Fixes #17

File: mack/__init__.py (hunk 1 above)
Comment (joao-fm-santos) [reply mack/__init__.py:428]: Apologies, you are correct :) Reverted

**Comment by joao-fm-santos** [comment]
Fixed comments and tomorrow will be adding some descriptions

**Comment by robertkossendey** [comment]
Thank you @joao-fm-santos! Could you make sure that the List[str] was not affected in other places as well? I think I saw 1-2 other places where the [str] was missing.

**Comment by MrPowers** [comment]
@robertkossendey - any chance you can just merge this one and then make those minor fixes?  This one seems pretty much done and might as well get it in there ;)

**Comment by MrPowers** [comment]
@joao-fm-santos - thanks for the contribution & thanks for pushing this over the finish line @robertkossendey
